Sources/Amplitude/Amplitude.m

```diff
@@ -1525,6 +1524,21 @@ - (void)addEventMiddleware:(id<AMPMiddleware> _Nonnull)middleware {
     [_middlewareRunner add:middleware];
 }
 
+/**
+ * The amount of time after an identify is logged that identify events will be batched before being uploaded to the server.
+ * The default is 30 seconds.
+ */
+-(BOOL)setIdentifyUploadPeriodSeconds:(int)uploadPeriodSeconds {
+    return [_identifyInterceptor setInterceptedIdentifyUploadPeriodSeconds:uploadPeriodSeconds];
```

Comment:
I end up keeping the class variable for this. When doing the integration test realized this could be called before initializing...

Reply:
@qingzhuozhen it doesn't seem necessary in my implementation. See tests, let me know if you think we still need this. Thanks

Reply:
The test there seems to call the initialize first, did you try to call it first before init with api key in integration test?

Reply:
@qingzhuozhen I added another test without calling `client.initializeWithApiKey()`. No errors. Let me know if you would still like me to update.